File: VB/SampleLayoutView/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraGrid.Views.Layout.Customization
Imports DevExpress.XtraEditors

Namespace SampleLayoutView
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			AddHandler myLayoutView1.ShowCustomization, AddressOf myLayoutView1_ShowCustomization
		End Sub

		Private Sub myLayoutView1_ShowCustomization(ByVal sender As Object, ByVal e As EventArgs)
			Dim btn = New SimpleButton()
			btn.Text = "New Button"
			btn.Location = New Point(250,9)
			myLayoutView1.CustomizationForm.Controls(0).Controls(0).Controls(1).Controls.Add(btn)
			btn.Select()
		End Sub
	End Class
End Namespace

File: CS/SampleLayoutView/Form1.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Windows.Forms;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraGrid.Views.Layout.Customization;
using DevExpress.XtraEditors;

namespace SampleLayoutView
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            myLayoutView1.ShowCustomization += new EventHandler(myLayoutView1_ShowCustomization);
        }

        void myLayoutView1_ShowCustomization(object sender, EventArgs e)
        {
            var btn = new SimpleButton();
            btn.Text = "New Button";
            btn.Location = new Point(250,9);
            myLayoutView1.CustomizationForm.Controls[0].Controls[0].Controls[1].Controls.Add(btn);
            btn.Select();
        }
    }
}
